### ISSUE TITLE
"_oldValue" does not need to be a SerializedField

### ISSUE THREAD

**Issue by soraphis** [unity-atoms/unity-atoms#387]
https://github.com/unity-atoms/unity-atoms/blob/5df498b41ee77de0c5faa73b76f7fd0ed890216a/Packages/Core/Runtime/Variables/AtomVariable.cs#L96-L97

probably the same goes for "_value". 
See: https://github.com/unity-atoms/unity-atoms/issues/146#issuecomment-1409735010

**Comment by toasterhead-master** [comment]
Interesting. But how would you see it if you want to through the inspector? And the same goes for _value. Plus, you won't be able to change it in the inspector anymore. So maybe what you are suggesting is adding 2 new values, _valueSerializable, and _oldValueSerializable, and they only exist to display the values. Plus the _valueSerializable would set the _value when you change it through the inspector. And that would solve the problem of them not resetting on playmode entered right?

**Comment by soraphis** [comment]
> But how would you see it through the inspector?

1. the main purpose of _oldValue is for onChangedEvent to provide the old value.
2. to see the oldValue even if its not a SerializedField: just turn the inspector into debug mode.
3. I'd argue that you're not supposed to change _oldValue in the inspector, anyways: a custom editor script could allow the modification of private variables, and could even show them without putting the inspector into debug mode.

>  So maybe what you are suggesting is adding 2 new values

No, definitely not. The idea is to reduce the amount of unnecessary serialized variables. There are already too much of them.

> Plus the _valueSerializable would set the _value

_initialValue fulfills this job just fine.


> And that would solve the problem of them not resetting on playmode entered right?

sadly, not necessarily. as it's hard to guarantee that a private field on a scriptable object will be reseted.

**Comment by toasterhead-master** [comment]
> _initialValue fulfills this job just fine.
But it doesn't raise an event when changed through the inspector, does it? But ok, with debug mode you can achieve the same result I guess. Or just draw them? I dunno, I've yet to be able to predict Unity's serializedfield or it's absence behaviour.

**Comment by soraphis** [comment]
> [_initialValue] ... But it doesn't raise an event when changed through the inspector, does it?

no, it does not. `_value` also did not. When in playmode the custom editor checks if `_value` was changed and then sets BaseValue, which sets Value which calls the events.

this can easily be done without _value field, and without _oldValue field, since it's only inspector magic anyways.

**Comment by rhys-vdw** [comment]
Just to second the above:

1. Yes you can show any value in an inspector, doesn't matter if it's serialized or not. You just need a custom editor window.
2. It won't fix #146 because the values will stick around after entering/leaving play mode because that has nothing to do with serialization. It will just stop the atom assets changing on disk after playing the game.
